system/jlib/jptree.ipp

```diff
@@ -1024,5 +1051,77 @@ public:
     }
 };
 
+class CPTreeEncodeNamesMaker : public CPTreeMaker
+{
+public:
+    CPTreeEncodeNamesMaker(byte flags=ipt_none, IPTreeNodeCreator *_nodeCreator=NULL, IPropertyTree *_root=NULL, bool _noRoot=false) : CPTreeMaker(flags, _nodeCreator, _root, _noRoot)
+    {
+    }
+    ~CPTreeEncodeNamesMaker()
+    {
+    }
+
+// IPTreeMaker
+    virtual void beginNode(const char *tag, bool arrayitem, offset_t startOffset) override
+    {
+        if (rootProvided)
+        {
+            currentNode = root;
+            rootProvided = false;
+        }
+        else
+        {
+            StringAttr encoded;
```

Comment:
As before, simpler as a StringBuffer?

Reply:
Same idea here.  I'll happily simplify the code after you comment on my question about what I though was a slight efficiency gain for 99% of cases.